### DISCUSSION TITLE
Concern regarding Consumer's metheod "consume"

### DISCUSSION THREAD

**Discussion by moatazelmohtaseb** [gocardless/hutch#378] [General]
In a project I'm working on, there's a concern regarding `Hutch::Consumer#consume` method:

```
class TestConsumer
  include Hutch::Consumer
  # consume >> won't be using the consume method
```

What happens is that all bindings to the `test_consumer` queue will be removed, even ones I've manually created. Which might be risky in some cases (human errors in dealing with the `consume` method).

**The questions here are following:** 
- What is the philosophy behind this behavior ?
- Are there any recommendations to mitigate this risk ? (which might accidentally cause the deletion of bindings)

**Comment by michaelklishin** [comment]
The fact that you don't use a specific method won't delete anything. Something else must be going on.
Consider putting together an executable way to reproduce and [inspecting what events RabbitMQ itself observes](https://rabbitmq.com/logging.html#internal-events).

Queues can be declared [as temporary](https://rabbitmq.com/queues.html#temporary-queues).

In general, if you don't expect a service like Hutch to manage the topology, you probably don't want to use it at all. Just build exactly what you need using Bunny directly.

**Reply by moatazelmohtaseb** [reply]
Hi @michaelklishin 

Thanks for the reply. 

I should've specified that the deletion (unbinding) only happens within the the topic exchange (default `hutch`).
The logic is done by the method `Hutch::Broker#unbind_redundant_bindings` where it unbinds the queue from any `routing_key` that isn't specified using the `Hutch::Consumer#consume` method.
